libdrizzle/conn.cc

```diff
@@ -1621,11 +1622,67 @@ static drizzle_return_t _setsockopt(drizzle_st *con)
     return DRIZZLE_RETURN_ERRNO;
   }
 
+  ret = 1;
```

Comment:
Yes. Keep alive are enabled as default. The freeze  problem seemed to be the default values in linux that were used before this PR: 
 
tcp_keepalive_time = 7200 (seconds)
tcp_keepalive_intvl = 75 (seconds)
tcp_keepalive_probes = 9 (number of probes)